core/src/main/java/jenkins/tasks/SimpleBuildWrapper.java

```diff
@@ -97,6 +148,17 @@ public void env(String key, String value) {
             }
             env.put(key, value);
         }
+        private Boolean wrapperRequiresWorkspace;
+        /**
+         * Keeps track of whether or not the specified wrapper requires a workspace context (working directory and
+         * launcher).
+         * <p>
```

Comment:
Should be either `@Restricted(Beta.class)` or otherwise make it clear that an implementation of `SimpleBuildWrapper` can ignore this.

Reply:
Not sure I follow. This is on `Context`, so an implementation of `SimpleBuildWrapper` generally does not need to care anyway.
And I'm not sure why the restriction to Beta would make sense; for example, `workflow-basic-steps` needs to use this.

Reply:
> This is on `Context`, so an implementation of `SimpleBuildWrapper` generally does not need to care anyway.

Implementations would call other methods on `Context`.

> for example, `workflow-basic-steps` needs to use this

And that is intended to be the _only_ example. `Beta` would just be a way of reminding other plugins not to use it.

Not important, just one suggestion to help clarify that this is part of the implementation, not the normal API.